src/common/CommandLine.cs

```diff
@@ -0,0 +1,1820 @@
+                _curPosition++;
+
+            object ret = null;
+            if (type.IsArray)
+            {
+                // Pass 1, Get the count
+                int count = 0;
+                int argPosition = _curPosition;
+                while (argPosition < _args.Count)
+                    if (_args[argPosition++] != null)
+                        count++;
+
+                if (count == 0 && isRequired)
+                    throw new CommandLineParserException("Required positional parameter " + name + " not present.");
+
+                Type elementType = type.GetElementType();
+                Array array = Array.CreateInstance(elementType, count);
+                argPosition = _curPosition;
+                int index = 0;
+                while (argPosition < _args.Count)
+                {
+                    string arg = _args[argPosition++];
+                    if (arg != null)
+                        array.SetValue(ParseValue(arg, elementType, name), index++);
+                }
+                _curPosition = _args.Count;
+                ret = array;
+            }
+            else if (_curPosition < _args.Count)     // A 'normal' positional parameter with a value
+            {
+                ret = ParseValue(_args[_curPosition++], type, name);
+            }
+            else // No value
+            {
+                if (isRequired)
+                    throw new CommandLineParserException("Required positional parameter " + name + " not present.");
+            }
+
+            return ret;
+        }
+        private bool DefineParameterSet(string name, string helpText)
+        {
+            Debug.Assert(_args != null);
+            if (_dashedParameterEncodedPositions == null)
+                ParseWordsIntoQualifiers();
+
+            if (!_paramSetEncountered && _positionalArgEncountered)
+                throw new CommandLineParserDesignerException("Positional parameters must not preceed parameter set definitions.");
+
+            _paramSetEncountered = true;
+            _positionalArgEncountered = false;               // each parameter set gets a new arg set   
+            _optionalPositionalArgEncountered = false;
+            if (_defaultParamSetEncountered)
+                throw new CommandLineParserDesignerException("The default parameter set must be defined last.");
+
+            bool isDefaultParameterSet = (name.Length == 0);
+            if (isDefaultParameterSet)
+                _defaultParamSetEncountered = true;
+
+            if (_mustParseHelpStrings)
+                AddHelp(new CommandLineParameter(name, null, helpText, typeof(bool), true, _noDashOnParameterSets, true));
+            if (_skipParameterSets)
+                return false;
+
+            // Have we just finish with the parameter set that was actually on the command line?
+            if (_parameterSetName != null)
+            {
+                _skipDefinition
```

Comment:
Ah yes it actually shows up correctly if I open the "Full View". And you're right, since this is compiled into the application now instead of being a separate library, that would actually work fine and return the correct name.